src/input.service.ts

```diff
@@ -455,9 +457,10 @@ export class InputService {
    * a connected gamepad somewhere.
    */
   private pollGamepad(now: number) {
-    navigator.getGamepads(); // refreshes all checked-out gamepads
+    const rawpads = navigator.getGamepads(); // refreshes all checked-out gamepads
 
     for (let i = 0; i < this.gamepads.length; i += 1) {
+      this.gamepads[i].pad = rawpads[i];
```

Comment:
You have to make sure that the gamepad is the one assigned to that specific instance. If someone disconnects a controller the indicies may change.